### PR TITLE
Add Networksets

### PR DESCRIPTION
**Pull request by lmm**
## Description

Add NetworkSets documentation with some docs fixes.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit d5d90fb7762714bf72c7b09c961a98e538847d98

https://deploy-preview-2487--calico.netlify.com

File: master/reference/calicoctl/resources/networkset.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+For `calicoctl` commands that specify a resource type on the CLI, the following
+aliases are supported (all case insensitive): `networkset`, `networksets`, `netsets`.
+
+### Sample YAML
+
+```yaml
+apiVersion: projectcalico.org/v3
+kind: NetworkSet
+metadata:
+  name: a-name-for-the-set
```
Comment (caseydavenport) [inline master/reference/calicoctl/resources/networkset.md]: I know we do this elsewhere, but I'd like to start using more realistic values here when providing sample yamls.

e.g.

```
name: database-ips
namespace: default
```

File: v3.1/reference/calicoctl/resources/index.md [hunk 2]
```diff
@@ -49,6 +49,8 @@ The following resources are supported:
 - [HostEndpoint]({{site.baseurl}}/{{page.version}}/reference/calicoctl/resources/hostendpoint)
 - [NetworkPolicy]({{site.baseurl}}/{{page.version}}/reference/calicoctl/resources/networkpolicy)
 - [GlobalNetworkPolicy]({{site.baseurl}}/{{page.version}}/reference/calicoctl/resources/globalnetworkpolicy)
+- [GlobalNetworkSet]({{site.baseurl}}/{{page.version}}/reference/calicoctl/resources/globalnetworkset)
```
Comment (caseydavenport) [inline v3.1/reference/calicoctl/resources/index.md]: hehe, good spot :)

File: v3.5/reference/calicoctl/resources/index.md [hunk 3]
```diff
@@ -49,6 +49,8 @@ The following resources are supported:
 - [HostEndpoint]({{site.baseurl}}/{{page.version}}/reference/calicoctl/resources/hostendpoint)
 - [NetworkPolicy]({{site.baseurl}}/{{page.version}}/reference/calicoctl/resources/networkpolicy)
 - [GlobalNetworkPolicy]({{site.baseurl}}/{{page.version}}/reference/calicoctl/resources/globalnetworkpolicy)
```
Comment (caseydavenport) [inline v3.5/reference/calicoctl/resources/index.md]: Do we need this change in v3.6 as well?

Comment (lmm) [reply v3.5/reference/calicoctl/resources/index.md]: Oops forgot to stage this

**Comment by caseydavenport** [comment]
If you could squash before merging that would be :ok_hand:

**Pull request merged**